From db49e1dada7c8a0d6a67008e06d425ac8fc1fa04 Mon Sep 17 00:00:00 2001
From: Simo Sorce <simo@redhat.com>
Date: Tue, 8 Jul 2025 06:45:35 -0400
Subject: [PATCH] Fix FIPS dispatcher table missing termination

On x86_64 this should have caused failures too, but we must have gotten
lucky that either a random zero value was present in the right position
to match function_id ...

Signed-off-by: Simo Sorce <simo@redhat.com>
---
 ossl/src/fips.rs | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/src/ossl/fips.rs b/src/ossl/fips.rs
index 67fbb02..25e30fa 100644
--- a/src/ossl/fips.rs
+++ b/src/ossl/fips.rs
@@ -690,6 +690,11 @@ static FIPS_PROVIDER: LazyLock<FipsProvider> = LazyLock::new(|| unsafe {
         dispatcher_struct!(args1; OSSL_FUNC_CRYPTO_SECURE_ALLOCATED; fips_secure_allocated),
         /* Indicator function */
         dispatcher_struct!(args2; OSSL_FUNC_INDICATOR_CB; fips_get_indicator_cb),
+        /* terminate table */
+        OSSL_DISPATCH {
+            function_id: 0,
+            function: None,
+        },
     ];
 
     let mut provider: *mut PROV_CTX = std::ptr::null_mut();
-- 
2.50.1

